roofit/roofitcore/src/RooMinimizer.cxx

```diff
@@ -159,29 +186,39 @@ void RooMinimizer::initMinimizerFcnDependentPart(double defaultErrorLevel)
    setErrorLevel(defaultErrorLevel);
 
    // Declare our parameters to MINUIT
-   _fcn->Synchronize(_theFitter->Config().ParamsSettings(), _fcn->getOptConst(), _verbose);
+   _fcn->Synchronize(_theFitter->Config().ParamsSettings(), _fcn->getOptConst(), _cfg.verbose);
 
    // Now set default verbosity
    if (RooMsgService::instance().silentMode()) {
       setPrintLevel(-1);
    } else {
       setPrintLevel(1);
    }
-}
-
 
-////////////////////////////////////////////////////////////////////////////////
-/// Destructor
+   // Set user defined and default _fcn config
+   execSetters();
+}
 
-RooMinimizer::~RooMinimizer()
+/// Execute all setters on _fcn
+void RooMinimizer::execSetters()
 {
-  if (_fcn) {
-    delete _fcn;
-  }
+   optimizeConst(_cfg.optConst);
```

Comment:
```suggestion
```
Similar to my other comment: I think this should not be called here, which is basically the RooMinimizer constructor. There was no `optimizeConst` called in the constructor so far, so doing this now would actually change the behavior of the code

Reply:
You also see from the name `optimizeConst` that this is not a simple configuration setter (it doesn't start with `set`, but it actually triggers some *action* that we don't necessarily want to do in the constructor